docs/source/programming.md

```diff
@@ -0,0 +1,223 @@
+# Programming a neutral-atom QPU
+
+Pulser enables to program Quantum Processing Units (QPUs) based on neutral atoms (also named _cold-atom Quantum Processing Units_). In this page, you will learn:
+
+- **What are the mathematical objects your are programming with Pulser ?** In quantum computing, the evolution of a quantum state is defined by an Hamiltonian. What is a quantum state in Pulser ? What are the Hamiltonians that can be defined in Pulser ?
+- **How do you program these mathematical objects ?** How do you define the quantum state in Pulser ? How do you define the Hamiltonian ? We will propose you a receipe to design your quantum program using Pulser. 
```

Comment:
nit: Some suggestions, take whichever ones you like
```suggestion
Pulser establishes a framework for programming Quantum Processing Units (QPUs) based on neutral atoms (also named _cold-atom Quantum Processing Units_). In this page, you will learn:

- **What mathematical objects encode a program in a neutral-atom QPU:** In quantum computing, the evolution of a quantum state is defined by an Hamiltonian. We define both these terms, as well as the level of control Pulser provides over each one.
- **How to program these mathematical objects with Pulser:** We give you a step-by-step guide on how to create a quantum program using Pulser.
```

Reply:
Also, the **bold** part of each bullet point could also link to the different sections in the page